test-data/unit/check-inference.test

```diff
@@ -935,6 +935,14 @@ if A:
 class A: pass
 class B: pass
 
+[case testAssigningAnyStrToNone]
+from typing import Tuple, TypeVar
+AnyStr = TypeVar('AnyStr', str, bytes)
+
+def f(x: AnyStr) -> Tuple[AnyStr]: pass
+x = None
+(x,) = f('')
+
```

Comment:
Can you add `reveal_type(x)` here to verify that we infer the correct type for `x`?